### PR TITLE
fileio: Fix compilation issue when SMACK is enabled

### PR DESCRIPTION
**Pull request by hoiho-amzn**

### CODE REVIEW

File: src/basic/fileio.c [hunk 1]
```diff
@@ -1429,6 +1429,7 @@ int rename_and_apply_smack_floor_label(const char *from, const char *to) {
                 return -errno;
 
 #if HAVE_SMACK_RUN_LABEL
+#include "smack-util.h"
```
Comment (bluca) [inline src/basic/fileio.c:1432]: please move this to the top of the file

Comment (hoiho-amzn) [reply src/basic/fileio.c:1432]: I did but LGTM was complaining `smack-util.h` not found - false positive?

Comment (mrc0mmand) [reply src/basic/fileio.c:1432]: I guess this is caused by `smack-util.h` being in `src/shared/`, and files from there shouldn't be included from stuff under `src/basic/` (see https://github.com/systemd/systemd/blob/main/docs/ARCHITECTURE.md#shared-code).

Comment (hoiho-amzn) [reply src/basic/fileio.c:1432]: right, problem with the new `fileio.c` is that it depends on `SMACK_FLOOR_LABEL` et al. which are defined in `smack-util.h`

Comment (bluca) [reply src/basic/fileio.c:1432]: yes indeed, I guess this whole function should move to smack-util.c (outside of the ifdef)

Comment (hoiho-amzn) [reply src/basic/fileio.c:1432]: Problem is that `rename_and_apply_smack_floor_label` is a two-step operation - it renames a file and then apply the SMACK floor label.

Should we just separate it into 2 functions and fix all other invocations of `rename_and_apply_smack_floor_label`?

Comment (bluca) [reply src/basic/fileio.c:1432]: I think it's fine to move the whole function, it's used only from tools

**Comment by poettering** [comment]
Yeah, this is a bug introduced by 494735f3d067bdeb97a5a98d09f9e2c4af705272, /cc @DaanDeMeyer 

The smack specific code should be moved to src/shared/. src/basic shouldn't try to link against code from there.

**Pull request closed without merging**